File: devpanellib/src/main/java/com/busylee/devpanel/info/ObjectInfo.java
package com.busylee.devpanel.info;

/**
 * Created by busylee on 16.10.15.
 */
public class ObjectInfo implements InfoEntry {

    private final Object mData;

    public ObjectInfo(Object data) {
        mData = data;
    }

    @Override
    public Object getData() {
        return mData;
    }
}

File: devpanellib/src/main/kotlin/com/busylee/devpanel/info/ObjectInfo.kt
package com.busylee.devpanel.info

/**
 * Created by busylee on 16.10.15.
 */
class ObjectInfo(
        override val data: Any,
        override val title: String,
        override val name: String = "") : InfoEntry<Any> {

    open class Builder(private val data: Any) {

        var title = ""

        open fun title(title: String): Builder {
            this.title = title
            return this
        }

        fun build(): ObjectInfo {
            return ObjectInfo(data, title)
        }

    }

}
